### ISSUE TITLE
Incomplete display of tiles in Mapbox when loading

### ISSUE THREAD

**Issue by reubengann** [SharpMap/SharpMap#52]
Hi, I'm using SharpMap.UI's Mapbox control. In my Form_Load, I load an MBTiles layer into the mapbox

```
MbTilesTileSource source = new MbTilesTileSource(sourcepath);
SharpMap.Layers.TileLayer layer = new SharpMap.Layers.TileLayer(source, "Map");
mapBox1.Map.BackgroundLayer.Add(layer);
mapBox1.Map.Zoom = 40000000;
mapBox1.Refresh();
```
The issue is that not all the tiles display, and there's a hole in the map. (I have attached a screenshot). If I call refresh at a later time (such as when the user clicks), the entire map displays without issue. When working with Brutile by itself (using their MBTiles example), I did not notice this issue.

Is there maybe an event I can hook into to know when the control is ready to render the whole viewport?

Thanks
![screenclip 14](https://cloud.githubusercontent.com/assets/20427974/24576204/9bf0da3c-1685-11e7-9c8a-b4c680778e94.png)

**Comment by PachisPachis** [comment]
Hi reubengann

I think your problem could be related to this other issue I reported:

https://github.com/SharpMap/SharpMap/issues/39

I noticed this behaviour also.

Regards.

**Comment by FObermaier** [comment]
Fixed with f178d6572412cd5dba340a0cf6fa10d0f1da435e
